File: Sponsors/mw-streams-java/src/main/java/com/mwam/hackathon/operators/Filter.java
package com.mwam.hackathon.operators;

import java.util.function.Function;

/**
 * Filters out elements of a stream that don't satisfy a predicate.
 */
public class Filter<T> extends Operator<T, T> {
    /**
     * Predicate to test events with.
     */
    private final Function<T, Boolean> testFn;

    public Filter(Function<T, Boolean> testFn) {
        super();
        this.testFn = testFn;
    }

    /**
     * Consumes some input and passes it to any children.
     *
     * @param input message to process
     */
    @Override
    public void consume(T input) {
        if (testFn.apply(input)) {
            emit(input);
        }
    }
}

File: Sponsors/mw_streams_python/mws/operators/filter.py
from typing import TypeVar, Callable

from mws.operators.operator import Operator


I = TypeVar("I")


class Filter(Operator[I, I]):
    def __init__(self, predicate: Callable[[I], bool]) -> None:
        super().__init__()
        self.predicate = predicate

    def consume(self, event: I) -> None:
        raise NotImplementedError("implement me")